### PR TITLE
[Merged by Bors] - feat: port Analysis.Normed.Order.Lattice

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Normed/Order/Lattice.lean [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b
+  solid : ∀ a b : α, |a| ≤ |b| → ‖a‖ ≤ ‖b‖
+#align normed_lattice_add_comm_group NormedLatticeAddCommGroup
+
+theorem solid {α : Type _} [NormedLatticeAddCommGroup α] {a b : α} (h : |a| ≤ |b|) : ‖a‖ ≤ ‖b‖ :=
+  NormedLatticeAddCommGroup.solid a b h
+#align solid solid
+
+instance Real.normedLatticeAddCommGroup : NormedLatticeAddCommGroup ℝ
+    where
```
Comment (dupuisf) [inline Mathlib/Analysis/Normed/Order/Lattice.lean:62]:
```suggestion
instance Real.normedLatticeAddCommGroup : NormedLatticeAddCommGroup ℝ where
```

File: Mathlib/Analysis/Normed/Order/Lattice.lean [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  ext1 x
+  simp only [Set.mem_preimage, Set.mem_singleton_iff, Set.mem_setOf_eq,
+    @neg_eq_zero_iff E _ _ (OrderedAddCommGroup.to_covariantClass_left_le E)  ]
+  -- porting note: I'm not sure why Lean couldn't synthesize this instance because it works with
+  -- `have : CovariantClass E E (· + ·) (· ≤ ·) := inferInstance`
+#align is_closed_nonneg isClosed_nonneg
+
+theorem isClosed_le_of_isClosed_nonneg {G} [OrderedAddCommGroup G] [TopologicalSpace G]
+    [ContinuousSub G] (h : IsClosed { x : G | 0 ≤ x }) : IsClosed { p : G × G | p.fst ≤ p.snd } :=
+  by
```
Comment (dupuisf) [inline Mathlib/Analysis/Normed/Order/Lattice.lean:213]:
```suggestion
    [ContinuousSub G] (h : IsClosed { x : G | 0 ≤ x }) : IsClosed { p : G × G | p.fst ≤ p.snd } := by
```

File: Mathlib/Analysis/Normed/Order/Lattice.lean [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  have : { p : G × G | p.fst ≤ p.snd } = (fun p : G × G => p.snd - p.fst) ⁻¹' { x : G | 0 ≤ x } :=
+    by
```
Comment (dupuisf) [inline Mathlib/Analysis/Normed/Order/Lattice.lean:215]:
```suggestion
  have : { p : G × G | p.fst ≤ p.snd } = (fun p : G × G => p.snd - p.fst) ⁻¹' { x : G | 0 ≤ x } := by
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**